File: 1769-minimum-number-of-operations-to-move-all-balls-to-each-box/1769-minimum-number-of-operations-to-move-all-balls-to-each-box.java
class Solution {
public int[] minOperations(String boxes) {
    int n = boxes.length();
    int[] answer = new int[n];
    int operations = 0;
    int count = 0;
 
    // Calculate the number of operations to move balls to the left of each box
    for (int i = 0; i < n; i++) {
        answer[i] += operations;
        count += boxes.charAt(i) - '0';
        operations += count;
    }

    // Calculate the number of operations to move balls to the right of each box
    operations = 0; count = 0;
     for (int i = n-1 ; i > -1; i--) {
       answer[i] += operations;
        count += boxes.charAt(i) - '0';
        operations += count;
    }

    return answer;
}
}

 

File: 1769-minimum-number-of-operations-to-move-all-balls-to-each-box/1769-minimum-number-of-operations-to-move-all-balls-to-each-box.kt
class Solution {
    fun minOperations(boxes: String): IntArray {
     val n = boxes.length
    val answer = IntArray(n)

    var count = boxes[0] - '0'
    var operations = 0
    for (i in 1 until n) {
        operations += count
        answer[i] = operations
        count += boxes[i] - '0'
    }

    count = boxes[n - 1] - '0'
    operations = 0
    for (i in n - 2 downTo 0) {
        operations += count
        answer[i] += operations
        count += boxes[i] - '0'
    }

    return answer
}

}